analyzer-core/src/main/java/ffb/analyzer/core/pubsub/Publisher.java

```diff
@@ -0,0 +1,79 @@
+package ffb.analyzer.core.pubsub;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class Publisher {
+    private Object mutex = new Object();
+
+    public enum Event {
+        EXAMPLE_EVENT
+    }
+
+    private static Publisher publisher;
+    private Map<Event, Set<Subscriber>> subscribers = new HashMap<>();
+
+    public static Publisher getPublisher() {
+        if (publisher == null) {
+            publisher = new Publisher();
+        }
+
+        return publisher;
+    }
+
+    public boolean unsubscribe(Event event, Subscriber subscriber) {
+        synchronized (mutex) {
+            final Set<Subscriber> eventSubscriptions = this.subscribers.get(event);
+
+            //TODO: Do we need to handle unsubscribe when not subscribed/event not listed?
+            return eventSubscriptions.remove(subscriber);
+        }
+    }
+
+    public boolean subscribe(Event event, Subscriber subscriber) {
+        boolean result;
+        synchronized (mutex) {
+            if (subscribers.containsKey(event)) {
```

Comment:
We should pre-initialize the HashSet during the construction of `Publisher`. It'll save us a lot of unnecessary checks if we can just add directly to the `Map`.

Reply:
Very good point. Updated to populate an empty HashSet in constructor.